packages/lab/src/list-next/useList.ts

```diff
@@ -28,45 +29,166 @@ export const useList = ({
   children,
   deselectable,
   multiselect,
+  displayedItemCount,
   onFocus,
 }: UseListProps) => {
   const listRef = useRef<HTMLUListElement | null>(null);
   const activeDescendantRef = useRef<string>("");
+  const selectedRef = useRef<number[]>([]);
   const [focusedIndex, setFocusedIndex] = useState<number | null>(null);
-  const [selectedIndexes, setSelectedIndexes] = useState<number[]>([]);
-  const [startRangeIndex, setStartRangeIndex] = useState(0);
+  const [startRangeIndex, setStartRangeIndex] = useState<number>(0);
+
+  const getAllOptions = (): Element[] => {
+    const list = listRef.current;
+    return (
+      Array.from(list?.children).filter(
+        (child) => child.getAttribute("role") === "option"
+      ) || []
+    );
+  };
+  const getAllActiveOptions = () =>
+    getAllOptions().filter(
+      (option) => option.getAttribute("aria-disabled") !== "true"
+    );
+
+  // debugger;
 
+  /**/
   const focusFirstItem = () => {
-    console.log(listRef.current?.children);
-    const firstItem = listRef.current.querySelector('[role="option"]');
+    // Find first active item
+    const firstItem = findNextOption(null);
     if (firstItem) {
-      focusItem(firstItem);
+      focusAndSelect(firstItem);
     }
   };
-  const focusLastItem = useCallback(() => {
-    const itemList = Array.from(
-      listRef.current.querySelectorAll('[role="option"]')
-    );
-    const lastItem = itemList[itemList.length - 1];
 
+  const focusLastItem = () => {
+    // Find last active item
+    const allActiveOptions = getAllActiveOptions();
+    const lastItem = allActiveOptions[allActiveOptions.length - 1];
     if (lastItem) {
-      focusItem(lastItem);
+      focusAndSelect(lastItem);
+    }
+  };
+
+  const getElementIndex = (option): number => {
```

Comment:
QQ - would the name `getItemIndex` be a better fit as it seems majority of functions refer to list items here as item, could keep it consistent